docs/proposals/29-autoscaling-storage-volumes.md

```diff
@@ -0,0 +1,326 @@
+  system's Prometheus instances.
+- Provide operational documentation for Gardener Operators.
+- Ability to enable/disable the feature per seed, via feature gate.
+
+### Non-Goals
+- Ability to activate autoscaling on a group of PVCs, other than specifying each PVC individually. Activate by selector
+  or namespace is not supported.
+- Ability to selectively deactivate autoscaling, e.g. by PVC selector or namespace.
+- Ability to scale a PVC if the storage class does not support resizing.
+- Ability to deactivate autoscaling for any of the observability volumes specified in the 'Goals' section.
+- Ability to scale volumes in shoot clusters.
+
+### Notes:
+- Shrinking a PVC is not supported. A limitation imposed by the underlying infrastructure, PVC size can only be
+  increased, and not decreased.
+- The newly introduced auto-scaling mode is predicated on the underlying infrastructure supporting volume resizing.
+  This ability is expressed through the K8s StorageClass type. The respective StorageClass must have
+  `allowVolumeExpansion: true`. For details, see the K8s volume expansion documentation.
+  
+## Proposal
+### PVC Autoscaler - Runtime Structure
+The overall runtime structure of the component is outlined in _Fig.1_:
+
+![01-runtime_structure.png](./assets/gep29-01-runtime_structure.png)
+
+_Fig.1: Runtime structure_
+
+For the role of a storage scaling controller, the existing, recently implemented Gardener PVC autoscaler application
+[[1]], hereafter referred to as `pvc-autoscaler`, is used. It is
+integrated as a seed system component, and runs as a deployment in the seed's `garden` namespace.
+Its primary driving signal is the PVC metrics from the seed's cache Prometheus instance, which it
+periodically examines, and if capacity is found to be near exhaustion, `pvc-autoscaler` takes action by updating the
+PVC's storage request.
+
+In this initial iteration, `pvc-autoscaler` scales two categories of `vali`, `prometheus`, and `alertmanager` volumes:
+those in shoot namespaces, and those in the seed's `garden` namespace.
+
+`pvc-autoscaler` publishes Prometheus metrics to anonymous scrapers via a `/metrics` HTTP endpoint. A `ServiceMonitor`
+object is created in the `garden` namespace, and drives the seed's `prometheus-operator` to configure a scrape on that
+endpoint by `prometheus-seed`.
+
+`pvc-autoscaler` runs in active/passive replication mode, based on the standard leader election mechanism, supplied by
+the K8s controller runtime library. Only the active replica is shown in _Fig.1_.
+
+### PVC Autoscaler - Deployer Structure
+The `pvc-autoscaler` application and its supporting artefacts are deployed by `gardener`, as part of the seed
+reconciliation flow.
+
+The `vali` and `prometheus` deployers, part of `gardenlet`'s shoot reconciliation flow, ensure that the PVC template section in
+the respective StatefulSet objects contains the annotations, necessary to instruct `pvc-autuscaler` how to act on the
```

Comment:
typo

Reply:
Fixed. Took me surprisingly long to spot.